packages/edit-site/src/components/sidebar/global-styles-v2/navigation/index.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { VIEW_ROOT } from './constants';
+import { GlobalStylesNavigationContext } from './context';
+
+export default function GlobalStylesNavigation( { children } ) {
+	const [ currentView, setCurrentView ] = useState( VIEW_ROOT );
+
+	return (
+		<GlobalStylesNavigationContext.Provider
+			value={ { currentView, setCurrentView } }
```

Comment:
´{ currentView, setCurrentView }` creates a new object every time the component rerenders, given that this is a context value I guess it will trigger unrequited rerender on the components tree. Maybe in this case useMemo is helpful.

Reply:
Yep, that was an oversight of mine 😉